docu/Concepts/TechnicalRequirements/UC_Introduction_of_Gradido-ID.md

```diff
@@ -0,0 +1,141 @@
+# Introduction of Gradido-ID
+
+## Motivation
+
+To introduce the Gradido-ID base on the requirement to identify an user account per technical key instead of using an email-address. Such a technical key ensures an exact identification of an user account without giving detailed information for possible missusage.
+
+Additionally the Gradido-ID allows to administrade any user account data like changing the email address or define several email addresses without any side effects on the identification of the user account.
+
+## Definition
+
+The formalized definition of the Gradido-ID can be found in the document [BenutzerVerwaltung#Gradido-ID](../BusinessRequirements/BenutzerVerwaltung#Gradido-ID).
+
+## Steps of Introduction
+
+To Introduce the Gradido-ID there are several steps necessary. The first step is to define a proper database schema with additional columns and tables followed by data migration steps to add or initialize the new columns and tables by keeping valid data at all.
+
+The second step is to decribe all concerning business logic processes, which have to be adapted by introducing the Gradido-ID.
+
+### Database-Schema
+
+#### Users-Table
+
+The entity users has to be changed by adding the following columns.
+
+| Column                   | Type   | Description                                                                            |
+| ------------------------ | ------ | -------------------------------------------------------------------------------------- |
+| gradidoID                | String | technical unique key of the user as UUID (version 4)                                   |
+| alias                    | String | a business unique key of the user                                                      |
```

Comment:
this is not applicable, because the user will have multiple aliases:
user <---1:n--> alias

a user might be reachable as `1234-5678-90` (UUID), `peter.lustig#1234` (his actual username containing some random value), `peter.lutig#1234` (his username with typo) and `slayer69#1234` (his former username).

It is to define how the alias is build and what are valid values, how long usernames are blocked once used by a user. How often he can change it. Furthermore it is to clarify if the random part for the username is needed (yes?!) and will the user carry that random part throughout username changes? (no!)

Reply:
As i understand Bernd's definition of an alias, that a user can set a free string as an alias. This must be unique in the community the user is member of.
- So a user can set his name like Ulf or Bernd or Moritz1 or Dario05 or something else..., if it is not used yet in the community
- there is no request for more than one alias per user
- a user can be identified by:
  * UUID (one time set and fix for ever)
  * alias (changeable)
  * email (changeable)
- there is no limit how often the user can change his alias
- if a blocked period after usage or before change will be necessary has to be researched, but currently i find no reason why

Reply:
A users alias must be stored "forever" tho(old transactions) and a user might be reachable under his old alias for a while - a more general approach with the alias might be useful here.